crates/router/src/core/pm_auth.rs

```diff
@@ -328,19 +322,39 @@ async fn store_bank_details_in_payment_methods(
         .customer_id
         .ok_or(ApiErrorResponse::CustomerNotFound)?;
 
+    #[cfg(all(
+        any(feature = "v1", feature = "v2"),
+        not(feature = "payment_methods_v2")
+    ))]
     let payment_methods = db
         .find_payment_method_by_customer_id_merchant_id_list(
+            &((&state).into()),
+            &key_store,
             &customer_id,
             merchant_account.get_id(),
             None,
         )
         .await
         .change_context(ApiErrorResponse::InternalServerError)?;
```

Comment:
Attach context messages wherever internal server errors are raised, here and elsewhere in this file.

Reply:
We'll take it up in a separate PM auth PR